### ISSUE TITLE
4 Taps issue with BTT2 and Wacom Intuos tablet

### ISSUE THREAD

**Issue by lorem-ndc** [fifafu/BetterTouchTool#681]
#### Description of bug/feature request/question:
I'm using BTT for years, mostly on my Wacom Intuos Pen an Touch, and also, on my MBPr's trackpad flawlessly.

Today i updated my actual BTT 1.93 version to the latest available, but i ran into an issue.

Since the v2.02 release, and through the actual 2.08 version, the 4 finger tap gesture is no more recognize by BTT on my tablet, and only on this device.

The 1, 2, 3 and 5 taps works as expected, only the 4 tap is gone. Other gestures as 2,3,4,5 fingers swipes work fine.

The MBPr's trackpad is not affected.

#### Affected input device (e.g. MacBook Trackpad, Magic Mouse/Trackpad, Touch Bar, etc.):
Wacom Intuos Pen and Touch Medium 2013 ( CTH-680 )

#### Device information:

Type of Mac: MacBookPro retina mid2014
macOS version: ElCapitan 10.11.16 ( don't want sierra for now )
BetterTouchTool version: 1.93

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):
I tried this workaround so far for the 2.08 :
- reset my gestures
- reset entirely BTT ( + appsupport folder and prefs ) and start anew
- create only the 2,3, 4, 5 taps gestures
- update my wacom drivers
- reset entirely my wacom prefs and config
- download previous 2.xx BTT versions
- download previous wacom drivers versions
- double check the mac's trackpad multitouch prefs
- reboots ...

I tried to use the old  6.3.15-3 and latest 6.3.20-11 driver with the BTT 2.02 and 2.08.

The intuos' multitouch has no issues, 4 fingers are detected using the wacom diagnostic.

Nothing fix the intuos' 4tap gesture, except to rollback and stick to the 1.93 version, which works just fine and dandy.

I hope it will help,
Best regards, in looking forward a solution or a fix in the next release of BTT
;)

**Comment by fifafu** [comment]
What kind of action have you assigned to the 4 finger tap? Does the action matter or doesn't it work with any action?

**Comment by lorem-ndc** [comment]
My 4 finger tap is assigned to a custom keyboard shortcut, but for the tests, I also tried with several BTT actions instead.
Neither of those methods worked with the intuos.

**Comment by ddiazescandon** [comment]
Same issue here.
Swiping 4 fingers is the default gesture for switching desktops, but now it became arrow left and right. Closing 4  fingers or opening gestures keep working.  I'm not sure if it is related because there is no single action in my btt for wacon,  I only use it for my leap motion, mouse and remote. I have close the app and restarted several times still the same.

**Comment by ddiazescandon** [comment]
Solved, update drivers or set to fabric mode an it will update and works.  Coincidentally the issue begin with the latest btt upgrade but it doesn't seem like an issue. It remain the same when btt was close.

But, there is a new weird bug, It just happen once but I had to restart BTT to deal with it.  Every time my phone was close  it began to sound like a wrong key several times. My phone was only on BLE with MacID app, BTT remote is on but it wasn't  open at all, I had to kill the app to stop the sound.

**Comment by liloo29** [comment]
Hey, how do you use BTT on your wacom? I have no option there. can you tell me please?
Thank you!

**Comment by ddiazescandon** [comment]
In a previous release there was  a dedicated feature to wacom, now you can just  call for a generic trackpad or table, right clicking your [gesture](http://i.imgur.com/HVk1lPt.png); I usually  use it for  screen shots with the BTT app, but I would wonder if we could get a drawing tool in it.

**Comment by liloo29** [comment]
Thanks, but I can't do anything with my wacom, even if I use it as a generic trackpad. it doesn't work at all. I have Intuos pen&touch. can you guide me? maybe I'm doing something wrong?

**Comment by ddiazescandon** [comment]
I have that one too. Wacom has in built gestures, you can activate or deactivate at preference cancel, those are pretty useful so I don't  recommend to re-mapping it in BTT, but you can use those gestures that wacom didn't use  and make your tablet even better. Like tap sequence and  five fingers  gestures.

**Comment by liloo29** [comment]
yes, I understand, but I don't get any response from my wacom with BTT. strange. I guess I'm doing something wrong. for [example](https://www.dropbox.com/s/ecpog0zu6v6j2am/Screen%20Shot%202017-04-09%20at%2023.18.10.png?dl=0).

**Comment by ddiazescandon** [comment]
http://i.imgur.com/zgNBZca.png 
BTT Version 2.071 (608) 
Wacom Into Pen & Touch Driver 6.3.21-15 

I actually try  your two fingers swipe right gesture and it works perfectly.

Maybe try reinstalling drivers forcing factory reset at the wacom app, or  plug-unplug.

**Comment by liloo29** [comment]
thanks, I just noticed you don't have the red text ["no multitouch capable trackpad found!"](https://www.dropbox.com/s/ddmr26k66xo1ory/Screen%20Shot%202017-04-10%20at%208.12.54.png?dl=0)
so I guess something is wrong with my driver (I have imac)

**Comment by ddiazescandon** [comment]
I'm using an iMac too. Red text also appeared on my BTT app but it disappear after creating the first gesture.

**Comment by lorem-ndc** [comment]
Hi fifafu !

Here a new feedback about my issue :

A couple of weeks ago, I gave new try with BTT2 with the update 2.099 with the same config as describe before ( macOs 10.11.6 and wacom 6.3.15-3 ).

Good news, the four fingers tap is back, everything seems to be alright now and works as usual !

I also checked BTT with the last wacom 6.3.21-8 driver, so far so good.

Today i'm on BTT 2.12 and wacom 6.3.22-3, no problems.

For my use and shortcuts, BTT2 is definitely fixed in this 2.12 since the 2.08 :)
